src/cpp/core/FileLockTests.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * FileLocktests.cpp
+ *
+ * Copyright (C) 2009-14 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include <core/FileLock.hpp>
+
+#include <core/Error.hpp>
+#include <core/FilePath.hpp>
+
+#include <boost/thread.hpp>
+
+#define RSTUDIO_NO_TESTTHAT_ALIASES
+#include <tests/TestThat.hpp>
+
+namespace rstudio {
+namespace core {
+namespace tests {
+
+TEST_CASE("File Locking")
+{
+   SECTION("A lock can only be acquired once")
+   {
+      Error error;
+      FilePath lockFilePath("/tmp/rstudio-tests/lock");
+      
+      FileLock lock1;
+      FileLock lock2;
+      
+      error = lock1.acquire(lockFilePath);
+      if (error)
+         LOG_ERROR(error);
+      
+      CHECK((error == Success()));
+      
+      error = lock2.acquire(lockFilePath);
+      if (error)
+         LOG_ERROR(error);
+      
+      CHECK_FALSE((error == Success()));
```

Comment:
Indeed, those are the semantics promised from the `file_lock` class (http://www.boost.org/doc/libs/1_59_0/doc/html/boost/interprocess/file_lock.html):

> A file lock, is a mutual exclusion utility similar to a mutex using a file. A file lock has sharable and exclusive locking capabilities and can be used with scoped_lock and sharable_lock classes. A file lock can't guarantee synchronization between threads of the same process so just use file locks to synchronize threads from different processes.

So using a `file_lock` to synchronize within a process may / won't work, as documented -- this is what the unit test erroneously checks.